### PR TITLE
handle acme-tls routes, override tls fields from vault, improve Route canonicalize

### PR DESCRIPTION
**Pull request by maorfr**

### CODE REVIEW

File: utils/openshift_resource.py [hunk 1]
```diff
@@ -113,6 +113,20 @@ def canonicalize(body):
                 body.get('type') == 'Opaque':
             body.pop('type')
 
+        if body['kind'] == 'Route':
+            if 'status' in body:
+                body.pop('status')
+            if body['spec'].get('wildcardPolicy') == 'None':
```
Comment (jmelis) [inline utils/openshift_resource.py]: This means that if wildcardPolicy is not explicitely set, it will be be set to `None`, right?

Comment (jmelis) [reply utils/openshift_resource.py]: I mean, that's what k8s will do, no?

File: utils/openshift_resource.py [hunk 2]
```diff
@@ -113,6 +113,20 @@ def canonicalize(body):
...
+                body['spec'].pop('wildcardPolicy')
+            # remove tls-acme specific params from Route
+            if 'kubernetes.io/tls-acme' in annotations:
+                annotations.pop(
+                    'kubernetes.io/tls-acme-awaiting-authorization-owner',
+                    None)
+                tls = body['spec']['tls']
```
Comment (jmelis) [inline utils/openshift_resource.py]: what if there is no `spec/tls` section? perhaps wrap this in a try except?

Comment (jmelis) [reply utils/openshift_resource.py]: I can imagine an unfortunate case where someone puts this in the resources folder, without the tls section and with that annotation.

File: utils/openshift_resource.py (hunk 1 above)
Comment (maorfr) [reply utils/openshift_resource.py]: right. same as ConfigMap's type `Opaque`.

File: utils/openshift_resource.py (hunk 2 above)
Comment (maorfr) [reply utils/openshift_resource.py]: for this kind of route (`kubernetes.io/tls-acme`) there is always a `tls` section:

https://github.com/app-sre/qontract-reconcile/pull/33/files#diff-c09ef4e4196ae9f15e156d3448ed7727R122

Comment (maorfr) [reply utils/openshift_resource.py]: added a test if `tls` is in `spec`

**Pull request merged**